chpl/chpl-service/src/test/java/gov/healthit/chpl/validation/listing/reviewer/ListingStatusAndUserRoleReviewerTest.java

```diff
@@ -0,0 +1,411 @@
+
+        CertifiedProductSearchDetails origListing = CertifiedProductSearchDetails.builder()
+                .id(1L)
+                .certificationEvent(getCertificationStatusEvent(1L, "01/01/2020", 1L, "Active"))
+                .certificationResult(CertificationResult.builder()
+                        .success(true)
+                        .criterion(CertificationCriterion.builder()
+                                .number("170.315 (a)(1)")
+                                .id(1L)
+                                .build())
+                        .build())
+                .build();
+        CertifiedProductSearchDetails updatedListing = CertifiedProductSearchDetails.builder()
+                .id(1L)
+                .certificationEvent(getCertificationStatusEvent(1L, "01/01/2020", 1L, "Active"))
+                .certificationEvent(getCertificationStatusEvent(1L, "02/01/2020", 2L, "Retired"))
+                .build();
+        updatedListing.setErrorMessages(new HashSet<String>());
+
+        reviewer.review(origListing, updatedListing);
+
+        assertEquals(1, updatedListing.getErrorMessages().size());
+    }
+
+    @Test
+    public void review_UserIsAcbAndListingUpdatedStatusFromRetiredToActiveAndCriteriaAdded_ErrorMessageAdded()
+            throws ParseException {
+        Mockito.when(resourcePermissions.isUserRoleAdmin()).thenReturn(false);
+        Mockito.when(resourcePermissions.isUserRoleOnc()).thenReturn(false);
+
+        CertifiedProductSearchDetails origListing = CertifiedProductSearchDetails.builder()
+                .id(1L)
+                .certificationEvent(getCertificationStatusEvent(1L, "01/01/2020", 1L, "Active"))
+                .certificationEvent(getCertificationStatusEvent(1L, "02/01/2020", 2L, "Retired"))
+                .build();
+        CertifiedProductSearchDetails updatedListing = CertifiedProductSearchDetails.builder()
+                .id(1L)
+                .certificationEvent(getCertificationStatusEvent(1L, "01/01/2020", 1L, "Active"))
+                .certificationEvent(getCertificationStatusEvent(1L, "02/01/2020", 2L, "Retired"))
+                .certificationEvent(getCertificationStatusEvent(1L, "03/01/2020", 1L, "Active"))
+                .certificationResult(CertificationResult.builder()
+                        .success(true)
+                        .criterion(CertificationCriterion.builder()
+                                .number("170.315 (a)(1)")
+                                .id(1L)
+                                .build())
+                        .build())
+                .build();
+        updatedListing.setErrorMessages(new HashSet<String>());
+
+        reviewer.review(origListing, updatedListing);
+
+        assertEquals(1, updatedListing.getErrorMessages().size());
+    }
+
+    @Test
+    public void review_UserIsAcbAndListingUpdatedStatusFromRetiredToActiveAndCriteriaDeleted_ErrorMessageAdded()
```

Comment:
Can we have one test of the form `review_UserIsAcbAndListingIsUpdatedWithAChangedCriteria_ErrorMessageAdded` to cover the state where a listing in a non-active state has a criteria added and removed in the same event? I had to read the reviewer to verify that case was being handled, but it'd be nice to see in a test if, for instance, a listing added a1 and removed b1 in the same activity